ddtrace/profiling/collector/_memalloc_tb.c

```diff
@@ -189,9 +228,19 @@ traceback_to_tuple(traceback_t* tb)
         PyTuple_SET_ITEM(frame_tuple, 1, PyLong_FromUnsignedLong(frame->lineno));
         PyTuple_SET_ITEM(frame_tuple, 2, frame->name);
         Py_INCREF(frame->name);
-        /* Class name */
-        PyTuple_SET_ITEM(frame_tuple, 3, empty_string);
-        Py_INCREF(empty_string);
+
+        PyObject* class_name;
+
+        if (frame->type) {
+            PyTypeObject* type = (PyTypeObject*)frame->type;
+            /* Improvement for Python 3.11+? */
+            // class_name = _PyType_GetQualName(type);
+            class_name = PyUnicode_FromString(type->tp_name);
+        } else {
+            class_name = empty_string;
```

Comment:
nit: what's the issue with setting this to `Py_None`?

Reply:
This is mostly to avoid having to handle the `None` vs `str` in the pprof encoding. Having everything as a string right away makes it easier to serialize everything in the end.